### DISCUSSION TITLE
Setup Documentation and help with class TFT_eSPI' has no member named 'getTouch' error

### DISCUSSION THREAD

**Discussion by ljunquera** [Bodmer/TFT_eSPI#2366] [Q&A - General] [answered]
I looked through the readme and reviewed [other sources](https://docs.lvgl.io/8.3/get-started/platforms/arduino.html#set-up-drivers), but I am still struggling to setup the environment. I was hoping there could be a more detailed explanation of how to setup the environment for Arduino. I understand you can use either User_Setup.h or User_Setup_Select.h, but how is it determined which one of these are used?

The [Tips](https://github.com/Bodmer/TFT_eSPI#tips) section did a good job of explaining in a little more detail.

My current problem is a` 'class TFT_eSPI' has no member named 'getTouch'` error, which I believe is related to `#define TOUCH_CS 21` based on [this post](https://github.com/Bodmer/TFT_eSPI/issues/1408#issuecomment-966500358), but I am not sure where I should do this and which file it might be in...or when it's running if the expected setup file is being used.

If I understand correctly, using the approach in Tips I should  have a User_Setup_Select.h with:

`#include <../TFT_eSPI_Setups/my_custom_setup.h>`

and the User_Setup.h line should be commented out like this:

`//#include <User_Setup.h>           // Default setup is root library folder`

based on this comment in the file:

`// Only ONE line below should be uncommented to define your setup.  Add extra lines and files as needed.`

and the User_Setup_Select.h file should be in the `/libraries/TFT_eSPI` folder. So the `#define TOUCH_CS 21` should be in the `my_custom_setup.h` file, correct?

I copied `Setup24_ST7789.h` to `/TFT_eSPI_Setups/my_custom_setup.h` but don't see `#define TOUCH_CS 21` in it, which I'm told should be `#define TOUCH_CS -1` in my case.

**Comment by Bodmer** [comment, marked as answer]
The[ User_Setup_Select.h file](https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup_Select.h) tells the compiler where to find the header that contains the setup information. By default it points to User_Setup.h, but it can be modified t point to one of the example setup files in the [User_Setups folder](https://github.com/Bodmer/TFT_eSPI/tree/master/User_Setups) or to a folder of your own making. All other setup file path lines must be commented out except the one you wish the compiler to find.

The simplest way forward is to see if there is an existing setup file that meets your needs or one that is close enough and can be altered. If the display does not have a touch screen then do not define TOUCH_CS

Provide information on the processor you are using, the display and interface type and I can provide information more relevant to your particular situation.
